echo.lic

```diff
@@ -4,15 +4,19 @@
 
 echo_action = proc do |server_string|
   open('echo_log_room.txt', 'a') do |f|
-    f.puts server_string
+    f.puts(server_string)
   end
   server_string
 end
+
+DownstreamHook.remove('echo_action')
 DownstreamHook.add('echo_action', echo_action)
 
-before_dying { DownstreamHook.remove('echo_action') }
+before_dying do
+  DownstreamHook.remove('echo_action')
+end
 
-loop do
-  pause 10
-  clear
+until script.gets.nil?
```

Comment:
Seems to me like there's a performance change with this code change. It looks like every script in lich is allocated a thread, and this is just to keep this "script" alive, even though it has no work to do (cause the real work is handled by the main thread in the DownstreamHook stuff)?

If that understanding is right, we switched from a thread that sleeps for 10 seconds, clears the scripts message queue, and then sleeps, to a thread that just spins pulling records off as fast as it can.

Can anyone verify if my understanding is right, and why we'd prefer one over the other?